test/e2e/daemon_set.go

```diff
@@ -0,0 +1,212 @@
+	_, err := c.DaemonSets(ns).Create(&expapi.DaemonSet{
+		ObjectMeta: api.ObjectMeta{
+			Name: simpleDSName,
+		},
+		Spec: expapi.DaemonSetSpec{
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: label,
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  simpleDSName,
+							Image: image,
+							Ports: []api.ContainerPort{{ContainerPort: 9376}},
+						},
+					},
+				},
+			},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Check that daemon pods launch on every node of the cluster.")
+	Expect(err).NotTo(HaveOccurred())
+	err = wait.Poll(retryInterval, retryTimeout, checkRunningOnAllNodes(f, label))
+	Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to start")
+
+	By("Stop a daemon pod, check that the daemon pod is revived.")
+	podClient := c.Pods(ns)
+
+	podList, err := podClient.List(labels.Set(label).AsSelector(), fields.Everything())
+	Expect(err).NotTo(HaveOccurred())
+	Expect(len(podList.Items)).To(BeNumerically(">", 0))
+	pod := podList.Items[0]
+	err = podClient.Delete(pod.Name, nil)
+	Expect(err).NotTo(HaveOccurred())
+	err = wait.Poll(retryInterval, retryTimeout, checkRunningOnAllNodes(f, label))
+	Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pod to revive")
+
+	complexDSName := "complex-daemon-set"
+	complexLabel := map[string]string{"name": complexDSName}
+	nodeSelector := map[string]string{"color": "blue"}
+	By(fmt.Sprintf("Creating daemon with a node selector %s", complexDSName))
+	_, err = c.DaemonSets(ns).Create(&expapi.DaemonSet{
+		ObjectMeta: api.ObjectMeta{
+			Name: complexDSName,
+		},
+		Spec: expapi.DaemonSetSpec{
+			Selector: complexLabel,
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: complexLabel,
+				},
+				Spec: api.PodSpec{
+					NodeSelector: nodeSelector,
+					Containers: []api.Container{
+						{
+							Name:  complexDSName,
+							Image: image,
+							Ports: []api.ContainerPort{{ContainerPort: 9376}},
+						},
+					},
+				},
+			},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	By(fmt.Sprintf("Initially, daemon pods should not be running on any nodes."))
+	err = wait.Poll(retryInterval, retryTimeout, checkRunningOnNoNodes(f, complexLabel))
+	Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pods to be running on no nodes")
+
+	By(fmt.Sprintf("Change label of node, check that daemon pod is launched."))
+	nodeClient := c.Nodes()
+	nodeList, err := nodeClient.List(labels.Everything(), fields.Everything())
+	Expect(len(nodeList.Items)).To(BeNumerically(">", 0))
+	nodeList.Items[0].Labels = nodeSelector
+	newNode, err := nodeClient.Update(&nodeList.Items[0])
+	Expect(err).NotTo(HaveOccurred())
+	Expect(len(newNode.Labels)).To(Equal(1))
+	err = wait.Poll(retryInterval, retryTimeout, checkDaemonPodOnNodes(f, complexLabel, []string{newNode.Name}))
+	Expect(err).NotTo(HaveOccurred(), "error waiting for daemon pods to be running on new nodes")
+}
```

Comment:
Can you add changing node selector to something that doesn't match, and verify the pod is killed? Should just be a few more lines of code.